q2_vsearch/tests/test_chimera.py

```diff
@@ -0,0 +1,202 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+
+import biom
+import numpy as np
+from qiime2.plugin.testing import TestPluginBase
+from qiime2.util import redirected_stdio
+from q2_types.feature_data import DNAFASTAFormat
+
+from q2_vsearch._chimera import (uchime_denovo, _uchime_denovo,
+                                 uchime_ref, _uchime_ref)
+from .test_cluster_features import _read_seqs
+
+
+class UchimeDenovoTests(TestPluginBase):
+
+    package = 'q2_vsearch.tests'
+
+    def setUp(self):
+        super().setUp()
+        input_sequences_fp = self.get_data_path('dna-sequences-3.fasta')
+        self.input_sequences = DNAFASTAFormat(input_sequences_fp, mode='r')
+        self.input_sequences_list = _read_seqs(self.input_sequences)
+
+        self.input_table = biom.Table(np.array([[100, 101, 103],
+                                                [99, 98, 99],
+                                                [4, 5, 6],
+                                                [2, 2, 2]]),
+                                      ['feature1', 'feature2', 'feature3',
+                                       'feature4'],
+                                      ['sample1', 'sample2', 'sample3'])
+
+    def test_uchime_denovo(self):
+        with redirected_stdio(stderr=os.devnull):
+            chime, nonchime, stats = uchime_denovo(
+                sequences=self.input_sequences, table=self.input_table)
+
+        obs_chime = _read_seqs(chime)
+        exp_chime = [self.input_sequences_list[3]]
+        self.assertEqual(obs_chime, exp_chime)
+
+        # sequences are reverse-sorted by abundance in output
+        obs_nonchime = _read_seqs(nonchime)
+        exp_nonchime = [self.input_sequences_list[0],
+                        self.input_sequences_list[1],
+                        self.input_sequences_list[2]]
+        self.assertEqual(obs_nonchime, exp_nonchime)
+
+        stats_text = stats.open().read()
```

Comment:
Can you use a context manager so the file is closed? There's a few more `open()` calls like this in the tests -- I'll note those with a quick comment about resource leaking to make them easy to find.

I've noticed a lot of tests under the `qiime2` org leak resources in this manner, and I've fixed them in a couple of plugins. Moving forward, it'd be a good idea to *always* close files because we've run up on file descriptor limits in the past, and programs with leaking resources now raise warnings in Python 3, which clutters up the test output (you'll see these warnings if you run the test file directly as a Python script).